pkg/util/validation/validate.go

```diff
@@ -177,6 +189,42 @@ func ValidateExemplar(userID string, ls []mimirpb.LabelAdapter, e mimirpb.Exempl
 	return nil
 }
 
+// ValidateHistograms returns an err if the histogram is invalid.
+// The returned error may retain the provided series labels.
+// It uses the passed 'now' time to measure the relative time of the histogram.
+func ValidateHistogram(now model.Time, cfg SampleValidationConfig, userID string, ls []mimirpb.LabelAdapter, h mimirpb.Histogram) ValidationError {
+	unsafeMetricName, _ := extract.UnsafeMetricNameFromLabelAdapters(ls)
+
+	if model.Time(h.Timestamp) > now.Add(cfg.CreationGracePeriod(userID)) {
+		DiscardedSamples.WithLabelValues(reasonTooFarInFuture, userID).Inc()
+		return newSampleTimestampTooNewError("histogram", unsafeMetricName, h.Timestamp)
+	}
+
+	checkSpans := func(sign string, spans []*mimirpb.BucketSpan, buckets []int64) error {
```

Comment:
Same comment as Prometheus about removing the `sign` from here and instead use errors.Wrap below with the sign